presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -490,14 +491,24 @@ protected Scope visitTable(Table table, Scope scope)
 
         Optional<ViewDefinition> optionalView = metadata.getView(session, name);
         if (optionalView.isPresent()) {
+            Statement statement = analysis.getStatement();
+            if (statement instanceof CreateView && ((CreateView) statement).isReplace() && ((CreateView) statement).getName().equals(table.getName())) {
```

Comment:
This check is not sufficient for names that are not fully qualified. It needs to compare both names after they are qualified via `createQualifiedObjectName()` (see the call a few lines above).